packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraExposureTests.m

```diff
@@ -5,51 +5,74 @@
 @import camera_avfoundation;
 @import XCTest;
 @import AVFoundation;
-#import <OCMock/OCMock.h>
 
-@interface FLTCam : NSObject <FlutterTexture,
-                              AVCaptureVideoDataOutputSampleBufferDelegate,
-                              AVCaptureAudioDataOutputSampleBufferDelegate>
-
-- (void)setExposurePointWithResult:(FlutterResult)result x:(double)x y:(double)y;
-@end
+#import "MockCaptureDeviceController.h"
+#import "MockDeviceOrientationProvider.h"
 
 @interface CameraExposureTests : XCTestCase
 @property(readonly, nonatomic) FLTCam *camera;
-@property(readonly, nonatomic) id mockDevice;
-@property(readonly, nonatomic) id mockUIDevice;
+@property(readonly, nonatomic) MockCaptureDeviceController *mockDevice;
+@property(readonly, nonatomic) MockDeviceOrientationProvider *mockDeviceOrientationProvider;
 @end
 
 @implementation CameraExposureTests
 
 - (void)setUp {
   _camera = [[FLTCam alloc] init];
-  _mockDevice = OCMClassMock([AVCaptureDevice class]);
-  _mockUIDevice = OCMPartialMock([UIDevice currentDevice]);
-}
+  _mockDevice = [[MockCaptureDeviceController alloc] init];
+  _mockDeviceOrientationProvider = [[MockDeviceOrientationProvider alloc] init];
 
-- (void)tearDown {
-  [_mockDevice stopMocking];
-  [_mockUIDevice stopMocking];
+  [_camera setValue:_mockDevice forKey:@"captureDevice"];
+  [_camera setValue:_mockDeviceOrientationProvider forKey:@"deviceOrientationProvider"];
```

Comment:
can you inject these 2 via `FLTCam`'s constructor? The string keyPath can be error prone

Reply:
Ah yes, absolutely, I think I did it in further commits in the initial PR, but I'll add DI for those here.